File: Weijia/69. Sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        c1=1
        c2=x
        while c1<=c2:
            mid=int((c2+c1)/2)
            s=int(x/mid)
            if s>mid:
                c1=mid+1
            elif s<mid:
                c2=mid-1
            else:
                return mid
        return c2

File: solution_zzj/69._Sqrt(x).cc
class Solution {
public:
    int mySqrt(int x) {
        if (x == 0 || x == 1)
            return x;
        long long l = 0, r = x / 2;
        while (l <= r) {
            long long m = (l + r) / 2;
            if (m * m == x) {
                return m;
            } else if (m * m > x) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }
        return r;
    }
};
